### PR TITLE
Fix macOS Github Actions

### LINKED ISSUE #1719: MacOS CI builds failing
Seems an updated Github Actions MacOS image has broken our CI

From a quick look, I noticed the following:
- Configure fails without any good info.
- pip fails and suggests using pipx or venv.

### PR DESCRIPTION
**Pull request by phprus**
Fixes #1719 

1) New macOS pipelines, based on
``macos-13`` - x86 tests
``macos-latest`` - ARM64 tests

2) Fix ``configure`` script (replaced unconditional add ``-std=c11``).

3) Switch to ``python`` ``venv``.

### CODE REVIEW

**Comment by phprus** [comment]
@Dead2, @nmoinvaz Review this PR please.

**Review by Dead2 — APPROVED** [review]
LGTM

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -546,21 +546,38 @@ jobs:
...
+            compiler: clang
+            cxx-compiler: clang++
+            cmake-args: -DWITH_SANITIZER=Address
+            codecov: macos_clang_arm64
+
+          - name: macOS GCC UBSAN
+            os: macos-13
             compiler: gcc-10
             cxx-compiler: g++-10
+            ldflags: -ld_classic
```
Comment (nmoinvaz) [inline .github/workflows/cmake.yml:568]: Why `-ld_classic`?

File: .github/workflows/cmake.yml [hunk 2]
```diff
@@ -546,21 +546,38 @@ jobs:
...
 
-          - name: macOS GCC UBSAN
+          - name: macOS Clang ASAN (ARM64)
             os: macos-latest
+            compiler: clang
+            cxx-compiler: clang++
+            cmake-args: -DWITH_SANITIZER=Address
+            codecov: macos_clang_arm64
+
+          - name: macOS GCC UBSAN
```
Comment (nmoinvaz) [inline .github/workflows/cmake.yml:562]: Maybe specify "GCC 10"?

Comment (phprus) [reply .github/workflows/cmake.yml:562]: Other pipelines do not show GCC version in it names.

Comment (nmoinvaz) [reply .github/workflows/cmake.yml:562]: Yeah, it looks like even for Clang. Ok. Nvm.

File: .github/workflows/cmake.yml (hunk 1 above)
Comment (phprus) [reply .github/workflows/cmake.yml:568]: Xcode 15 uses a new linker that is not compatible with GCC.
``-ld_classic`` switches to the old linker.

Homebrew ``gcc@11`` and later have a built-in workaround for this issue:

https://github.com/Homebrew/homebrew-core/blob/c0218d50084e300cba26da84028acfd4917ce623/Formula/g/gcc%4011.rb#L92-L96

Comment (nmoinvaz) [reply .github/workflows/cmake.yml:568]: Oh wow. Ok. Can you add a comment in the yaml about that?

Comment (phprus) [reply .github/workflows/cmake.yml:568]: PR updated.

**Review by nmoinvaz — APPROVED** [review]

**Pull request merged**